website/src/components/home/CanvasList.jsx

```diff
@@ -98,7 +98,15 @@ class CanvasList extends React.Component {
     // Listen for any new canvases that might be added.
     firebase.database().ref('/users').child(userId)
       .child(RC.CANVASES).on('child_added', (canvasSnap) => {
-        this.fetchCanvasInfo(canvasSnap.key);
+        if (canvasSnap.val()) {
+          this.fetchCanvasInfo(canvasSnap.key);
+        }
+      });
+
+    // Listen for any canvases that might be removed.
+    firebase.database().ref('/users').child(userId)
+      .child(RC.CANVASES).on('child_changed', (canvasSnap) => {
+        this.props.dispatch(CanvasActions.removeCanvas(canvasSnap.key));
```

Comment:
it would be a good idea to ensure that the canvasSnap's vlaue is false before dispatching a remove